app/controllers/chapters_controller.rb

```diff
@@ -267,6 +265,23 @@ def post_chapter
     flash[:notice] = ts('Chapter has been posted!')
   end
 
+  # TEMPORARY hack-like thing to fix the fact that chaptered works weren't
+  # hit-counted or added to history at all. Increases hit count if you are
+  # - accessing the first chapter
+  # - accessing the most recent chapter if you are
+  # -- coming without a referer, e.g. via a subscription email
+  # -- coming through the recent chapter link, but first passing through the
+  #    adult content warning
+  # -- coming from anywhere else that is not the work, e.g. social media link
+  def increase_work_hit_count?
+    chapter_position = @chapters.index(@chapter)
+    chapter_position.zero? ||
+      chapter_position == (@work.number_of_posted_chapters - 1) &&
+        (request.referer.nil? ||
+        request.referer.match(/#{chapter_path(@chapter)}/) && params[:view_adult] ||
```

Comment:
One last change: similar to https://github.com/otwcode/otwarchive/pull/2779#discussion_r110215022, we should add `\b` to the end of the chapter path regex.